# HG changeset patch
# User Pantxo Diribarne <pantxo.diribarne@gmail.com>
# Date 1492977709 -7200
#      Sun Apr 23 22:01:49 2017 +0200
# Node ID 39045e50ea450eda7253c75725b20c6ee721c890
# Parent  32d532b8e7d0ca6fd194539f19209d66546698d9
Fix regression with true color unint8 images (bug #50797)

* graphics.cc (array_property::validate): use more generic NDArray type for
   constraints checking

diff --git a/libinterp/corefcn/graphics.cc b/libinterp/corefcn/graphics.cc
--- a/libinterp/corefcn/graphics.cc
+++ b/libinterp/corefcn/graphics.cc
@@ -1458,17 +1458,17 @@ array_property::validate (const octave_v
                     break;
                 }
             }
         }
     }
 
   if (xok)
     {
-      Matrix v_mat = v.matrix_value ();
+      NDArray v_mat = v.array_value ();
       // Check min and max
       if (! octave::math::isnan (minval.first))
         {
           for (octave_idx_type i = 0; i < v_mat.numel (); i++)
             if (minval.second && minval.first > v_mat(i))
               error ("set: \"%s\" must be greater than or equal to %g",
                      get_name ().c_str (), minval.first);
             else if (! minval.second && minval.first >= v_mat(i))
